### PR TITLE
Add metrci names/labels validation

### PR DESCRIPTION
**Pull request by mehdi-aouadi**
## PR Description
Adds metrics name and label validation.
Metric name must match regex `[a-zA-Z_:][a-zA-Z0-9_:]*`
Metric labels must match regex `[a-zA-Z_][a-zA-Z0-9_]*`

## Fixed Issue(s)

Fixes https://github.com/ConsenSys/teku-internal/issues/11

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: infrastructure/metrics/src/test/java/tech/pegasys/teku/infrastructure/metrics/StubMetricSystemTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    Assertions.assertThrows(
+        IllegalArgumentException.class,
+        () ->
+            metricsSystem.createLabelledCounter(
+                metricCategory, "$incorrect_name", "", "correct_label"));
+    Assertions.assertThrows(
+        IllegalArgumentException.class,
+        () ->
+            metricsSystem.createLabelledCounter(
+                metricCategory, "incorrect-name", "", "correct_label"));
```
Comment (rolfyone) [inline infrastructure/metrics/src/test/java/tech/pegasys/teku/infrastructure/metrics/StubMetricSystemTest.java]: probably using a parameterized test for this

File: infrastructure/metrics/src/test/java/tech/pegasys/teku/infrastructure/metrics/StubMetricSystemTest.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  @Test
+  public void mustAcceptValidLabelNames() {
+    metricsSystem.createLabelledCounter(metricCategory, "correct_name", "", "correctlabel");
+    metricsSystem.createLabelledCounter(metricCategory, "correct_name", "", "correctLabel");
+    metricsSystem.createLabelledCounter(metricCategory, "correct_name", "", "correct_label");
+    metricsSystem.createLabelledCounter(metricCategory, "correct_name", "", "_correct_label");
```
Comment (rolfyone) [inline infrastructure/metrics/src/test/java/tech/pegasys/teku/infrastructure/metrics/StubMetricSystemTest.java]: even these i think would be easier to see as parameterized, just feeding in the string under test...

File: infrastructure/metrics/src/test/java/tech/pegasys/teku/infrastructure/metrics/StubMetricSystemTest.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+  @Test
+  public void mustAcceptValidMetricNames() {
+    metricsSystem.createLabelledCounter(metricCategory, "correctname", "", "correct_label");
+    metricsSystem.createLabelledCounter(metricCategory, ":correctname", "", "correct_label");
+    metricsSystem.createLabelledCounter(metricCategory, "_correctname", "", "correct_label");
+    metricsSystem.createLabelledCounter(metricCategory, "correctNAME", "", "correct_label");
+    metricsSystem.createLabelledCounter(metricCategory, "correct_name", "", "correct_label");
+    metricsSystem.createLabelledCounter(metricCategory, "correct_name__", "", "correct_label");
+    metricsSystem.createLabelledCounter(metricCategory, "correct_name::123", "", "correct_label");
```
Comment (rolfyone) [inline infrastructure/metrics/src/test/java/tech/pegasys/teku/infrastructure/metrics/StubMetricSystemTest.java]: parameterised again.. should make it easy when things break too..make one start with a capital, any really

**Comment by rolfyone** [comment]
Looks pretty similar to what I was expecting :) mostly just cleanup tests and I think we're good...

**Review by ajsutton — APPROVED** [review]
LGTM.

**Pull request merged**